src/Library/demos/HTTP Request/main.vala

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env -S vala workbench.vala --pkg libadwaita-1 --pkg libsoup-3.0 --pkg json-glib-1.0
+
+public errordomain FetchError {
+  FAILED_REQUEST,
+  FAILED_TO_PARSE
+}
+
+private Gtk.TextView article_text_view;
+private Gtk.Label article_title;
+
+public async void main () {
+  article_text_view = (Gtk.TextView) workbench.builder.get_object ("article_text_view");
+  article_title = (Gtk.Label) workbench.builder.get_object ("article_title");
+  try {
+    yield fetch_wikipedia_todays_featured_article ();
+  } catch (Error e) {
+    critical (e.message);
+  }
+}
+
+private async void fetch_wikipedia_todays_featured_article () throws Error {
+  var date = new DateTime.now_local ();
+  var http_session = new Soup.Session ();
+
+  // https://api.wikimedia.org/wiki/Feed_API/Reference/Featured_content
+  string language = "en";
+  string url =
+    @"https://api.wikimedia.org/feed/v1/wikipedia/$language/featured/$(date.format ("%Y/%m/%d"))";
+
+  var message = new Soup.Message ("GET", url);
+  InputStream input_stream = yield http_session.send_async (message, Priority.DEFAULT, null);
```

Comment:
I think its way easier to use `send_and_read_async()` than using an `InputStream`. Was there a specific reason to do so?

Reply:
Just for consistency with the JS demo, as it uses `send_async` too and reads the response using an `InputStream`, but yeah, `send_and_read_async` is way easier and probably what should be done in the first place. I will use `send_and_read_async` and send a PR to use it in the JS demo later.

Reply:
Thanks! That is of course the best solution :)